File: Lib/FPVR/Interfaces/IGimbalInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FPVR
{
    /// <summary>
    /// The interface for reading gimbal state.
    /// </summary>
    public interface IGimbalInfo
    {
        /// <summary>
        /// Gets the current attitude of the gimbal.
        /// </summary>
        /// <value>
        /// The current attitude of the gimbal.
        /// </value>
        Attitude Attitude { get; }

        /// <summary>
        /// Raised when the value of the <see cref="Attitude"/> property has changed.
        /// </summary>
        public event EventHandler AttitudeChanged;
    }
}

File: Android/FPVR/app/src/main/java/com/solersoft/fpvr/fpvrlib/IGimbalInfo.java
package com.solersoft.fpvr.fpvrlib;

/**
 * The interface for reading gimbal state
 */
public interface IGimbalInfo
{
    //region Public Methods
    /**
     * Register a callback to be invoked when gimbal values have changed.
     * @param l The callback that will run
     */
    public void setGimbalListener(GimbalListener l);
    //endregion

    //region Public Properties
    /**
     * Gets the current attitude of the gimbal.
     * @return The current attitude of the gimbal.
     */
    public Attitude getAttitude();

    /**
     * Gets the attitude capabilities of the gimbal.
     * @return An {@link AttitudeCapabilities} that describes the capabilities of the gimbal.
     */
    public AttitudeCapabilities getAttitudeCapabilities();
    //endregion
}
